### PR TITLE
[HAProxy] set event.module and event.dataset

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR adds `event.module` and `event.dataset` to the integration.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

### CODE REVIEW

File: packages/haproxy/manifest.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ categories:
   - web
 release: experimental
 conditions:
-  kibana.version: "^7.13.0"
+  kibana.version: "^7.14.0"
```
Comment (jsoriano) [inline packages/haproxy/manifest.yml:18]: Why is this change needed? Wouldn't be better if users of 7.13 can already benefit from this change?

Comment (mtojek) [reply packages/haproxy/manifest.yml:18]: I think we agreed before not to release changes for older stacks if it isn't necessary. This way changes are added only the latest one.

@andresrc I hope that's ok?

**Review by jsoriano — APPROVED** [review]

File: packages/haproxy/manifest.yml (hunk 1 above)
Comment (jsoriano) [reply packages/haproxy/manifest.yml:18]: Ok, let's continue like this if this is the current convention, but I think this artificially couples packages to versions of the stack, ideally packages should depend on the minor version of the stack that they require to work. Changes that don't affect compatibility with the stack shouldn't need to update this.
